File: s04e01_vetores/vetores.cpp
#include <iostream>
#include <sstream>
#include <vector>

using namespace std;

void show(vector<int> vet){
    cout << "[ ";
    for(size_t i = 0; i < vet.size(); i++)
        cout << vet[i] << " ";
    cout << "]\n";
}

int main(){
    string line;
    vector<int> vet;
    while(true){
        getline(cin, line);
        stringstream ss(line);
        string cmd;
        ss >> cmd;
        if(cmd == "end"){
            break;
        }else if(cmd == "add"){
            int value;
            while(ss >> value){
                vet.push_back(value);
            }
        }else if(cmd == "show"){
            show(vet);
        }else{
            cout << "fail: comando invalido\n";
        }
    }

    return 0;
}

File: s04e01_vetores/Vetores.java
import java.util.Scanner;
import java.util.ArrayList;

public class Vetores {

    static void show(ArrayList<Integer> vet){
        System.out.print("[ ");
        for(int i = 0; i < vet.size(); i++)
            System.out.print("" + vet.get(i) + " ");
        System.out.println("]");
    }

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        ArrayList<Integer> vet = new ArrayList<Integer>();

        while(true){
            String line = scanner.nextLine();
            String[] ui = line.split(" ");
            if(ui[0].equals("end")){
                break;
            }else if(ui[0].equals("add")){
                for(int i = 1; i < ui.length; i++)
                    vet.add(Integer.parseInt(ui[i]));
            }else if(ui[0].equals("show")){
                Vetores.show(vet);   
            }else{
                System.out.println("fail: comando invalido");
            }
        }
        scanner.close();
    }
}